### DISCUSSION TITLE
Appium noReset:true not working

### DISCUSSION THREAD

**Discussion by Ancybaby** [appium/.github#9] [Q&A]
Hi Team,

I'am trying to automate android mobile application testing using Appium and TestComplete. I donot want to reinstall the application everytime and thus I'am passing noReset:true capability to Appium server.

Even though the noReset:true capability is passed , I see the below logs in Appium where appium tries to reset the application data 

Performing fast reset on 'com.app'
Running adp -p 5037 -s devicename shell am force-stop "app"
Running adb -P 5037 -s devicename shell pm clear "app"
performed fast rest on the installed application(stop and clear)**_****_

I also tried passing fullReset:false and skipDeviceinitialization:true . nothing works and application is getting reset sometimes. It does work at times. I have observed that it resets data after 4 rounds of testing with the same application

How do I stop Appium from resetting application data even after mutiple rounds of testing with the same application instance?

**Comment by Ancybaby** [comment]
Appium Logs :
******************************

[HTTP] --> POST /session {"capabilities":{"firstMatch":[{"appium:app":"/users/svc-pspc-devops/BeOnPTT.apk","appium:appActivity":"com.harris.rf.beonptt.android.ui.MainActivity","appium:appPackage":"com.harris.rf.beonptt.android.ui","appium:automationName":"UIAutomator2","appium:bitbar_findDevice":false,"appium:deviceName":"Galaxy XCover6 Pro","appium:nativeWebScreenshot":true,"appium:newCommandTimeout":"300","appium:noReset":true,"appium:platformVersion":"14","platformName":"android"}]}}
[AppiumDriver@eb48] Calling AppiumDriver.createSession() with args: [null,null,{"firstMatch":[{"appium:app":"/users/svc-pspc-devops/BeOnPTT.apk","appium:appActivity":"com.harris.rf.beonptt.android.ui.MainActivity","appium:appPackage":"com.harris.rf.beonptt.android.ui","appium:automationName":"UIAutomator2","appium:bitbar_findDevice":false,"appium:deviceName":"Galaxy XCover6 Pro","appium:nativeWebScreenshot":true,"appium:newCommandTimeout":"300","appium:noReset":true,"appium:platformVersion":"14","platformName":"android"}]}]
[AppiumDriver@eb48] Event 'newSessionRequested' logged at 1722536071686 (14:14:31 GMT-0400 (Eastern Daylight Time))
info Appium Attempting to find matching driver for automationName 'UIAutomator2' and platformName 'android'
info Appium The 'uiautomator2' driver was installed and matched caps.
info Appium Will require it at /Users/svc-pspc-devops/.appium/node_modules/appium-uiautomator2-driver
dbug Appium Requiring driver at /Users/svc-pspc-devops/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[AppiumDriver@eb48] Appium v2.4.0 creating new AndroidUiautomator2Driver (v2.40.2) session
[AppiumDriver@eb48] Checking BaseDriver versions for Appium and AndroidUiautomator2Driver
[AppiumDriver@eb48] Appium's BaseDriver version is 9.10.3
[AppiumDriver@eb48] AndroidUiautomator2Driver's BaseDriver version is 9.10.3
[AndroidUiautomator2Driver@2069] 
[AndroidUiautomator2Driver@2069] Creating session with W3C capabilities: {
  "alwaysMatch": {
    "platformName": "android",
    "appium:app": "/users/svc-pspc-devops/BeOnPTT.apk",
    "appium:appActivity": "com.harris.rf.beonptt.android.ui.MainActivity",
    "appium:appPackage": "com.harris.rf.beonptt.android.ui",
    "appium:automationName": "UIAutomator2",
    "appium:bitbar_findDevice": false,
    "appium:deviceName": "Galaxy XCover6 Pro",
    "appium:nativeWebScreenshot": true,
    "appium:newCommandTimeout": "300",
    "appium:noReset": true,
    "appium:platformVersion": "14"
  },
  "firstMatch": [
    {}
  ]
}


Logs where it force stop the application and reset it 
********************************************************

[ADB] Getting install status for com.harris.rf.beonptt.android.ui
[ADB] Running '/Users/svc-pspc-devops/Library/Android/sdk/platform-tools/adb -P 5037 -s R5CW918E5WK shell pm path com.harris.rf.beonptt.android.ui'
[ADB] 'com.harris.rf.beonptt.android.ui' is installed
[ADB] Running '/Users/svc-pspc-devops/Library/Android/sdk/platform-tools/adb -P 5037 -s R5CW918E5WK shell am force-stop com.harris.rf.beonptt.android.ui'
[HTTP] --> GET /session/c480e687-5e55-40c5-bec5-967738474d70/screenshot {}
[AndroidUiautomator2Driver@2069] Calling AppiumDriver.getScreenshot() with args: ["c480e687-5e55-40c5-bec5-967738474d70"]
[ADB] Running '/Users/svc-pspc-devops/Library/Android/sdk/platform-tools/adb -P 5037 -s R5CW918E5WK shell pm clear com.harris.rf.beonptt.android.ui'
[AndroidDriver] Performed fast reset on the installed 'com.harris.rf.beonptt.android.ui' application (stop and clear)
[ADB] Running '/Users/svc-pspc-devops/Library/Android/sdk/platform-tools/adb -P 5037 -s R5CW918E5WK shell am start -W -n com.harris.rf.beonptt.android.ui/com.harris.rf.beonptt.android.ui.MainActivity -S'

**Comment by mykola-mokhnach** [comment]
Please try the same with the most recent driver and server version

**Comment by Ancybaby** [comment]
Upgrading driver helped. Thanks
